RR.App.Tests/Controllers/HRIS/EmployeeDataControllerUnitTests.cs

```diff
@@ -1,115 +1,133 @@
+        
+        _employeeDataDto = EmployeeDataTestData.EmployeeDataOne.ToDto();
     }
 
-    [Fact(Skip = "Current user needs to be set for validations on endpoint")]
+    [Fact]
     public async Task GetEmployeeDataReturnsOkResult()
     {
-        _employeeDataServiceMock.Setup(service => service.GetAllEmployeeData(1))
+        _identity.Setup(i => i.Role).Returns("Admin");
+        _identity.SetupGet(i => i.EmployeeId).Returns(2);
+
+        _employeeDataServiceMock.Setup(x => x.GetAllEmployeeData(_employeeDataDto.EmployeeId))
                                .ReturnsAsync(_employeeDataDtoList);
 
-        var result = await _controller.GetEmployeeData(1);
+        var result = await _controller.GetEmployeeData(_employeeDataDto.EmployeeId);
 
         var okResult = Assert.IsType<OkObjectResult>(result);
-        var model = Assert.IsType<List<EmployeeDataDto>>(okResult.Value);
-        _employeeDataServiceMock.Verify(service => service.GetAllEmployeeData(1), Times.Once);
+        var returnValue = Assert.IsType<List<EmployeeDataDto>>(okResult.Value);
+        Assert.Equal(_employeeDataDtoList, returnValue);
+        _employeeDataServiceMock.Verify(service => service.GetAllEmployeeData(_employeeDataDto.EmployeeId), Times.Once);
     }
 
-    [Fact(Skip = "Current user needs to be set for validations on endpoint")]
-    public async Task GetEmployeeDataReturnsNotFoundResult()
+    [Fact]
+    public async Task GetEmployeeUnauthorizedAccess()
     {
-        _employeeDataServiceMock.Setup(service => service.GetAllEmployeeData(1))
-                               .ReturnsAsync((List<EmployeeDataDto>?)null);
+        _identity.Setup(i => i.Role).Returns("Developer");
+        _identity.SetupGet(i => i.EmployeeId).Returns(5);
 
-        var result = await _controller.GetEmployeeData(1);
+        var result = await MiddlewareHelperUnitTests.SimulateHandlingExceptionMiddlewareAsync(async () => await _controller.GetEmployeeData(2));
 
         var notFoundResult = Assert.IsType<NotFoundObjectResult>(result);
-        var errorMessage = Assert.IsType<string>(notFoundResult.Value);
-        Assert.Equal("Employee data not found", errorMessage);
-        _employeeDataServiceMock.Verify(service => service.GetAllEmployeeData(1), Times.Once);
+        Assert.Equal(StatusCodes.Status404NotFound, notFoundResult.StatusCode);
+        Assert.Equal("User data being accessed does not match user making the request.", notFoundResult.Value);
     }
 
-    [Fact(Skip = "Current user needs to be set for validations on endpoint")]
-    public async Task SaveEmployeeDataReturnsOkResult()
+    [Fact]
+    public async Task SaveUnauthorizedAccess()
     {
+        _identity.Setup(i => i.Role).Returns("Developer");
+        _identity.SetupGet(i => i.EmployeeId).Returns(5);
+
+        var newController = new EmployeeDataController(new AuthorizeIdentityMock(), _employeeDataServiceMock.Object);
```

Comment:
Refactor: the controller is already initialized on line 24. There is no need to do it again. We can just use __controller instead of newController. Double check the rest of the function as well.

Reply:
Fixed